tests/sensors/test_base_sensor.py

```diff
@@ -60,12 +61,12 @@ def setUp(self):
             'start_date': DEFAULT_DATE
         }
         self.dag = DAG(TEST_DAG_ID, default_args=args)
+        self.dag.sync_to_db()
 
-        session = settings.Session()
-        session.query(TaskReschedule).delete()
-        session.query(DagRun).delete()
-        session.query(TaskInstance).delete()
-        session.commit()
+        with create_session() as session:
+            session.query(TaskReschedule).delete()
+            session.query(DagRun).delete()
+            session.query(TaskInstance).delete()
```

Comment:
who handles the commit? Is the context_manager for create_session handling the commit in __exit__ ?

Reply:
The `provide_session` uses, `create_session` which will `.commit()` after the yield: https://github.com/apache/airflow/blob/6b38649fa6cdf16055c7f5458050c70f39cac8fd/airflow/utils/db.py#L44